tracer/src/Datadog.Trace.ClrProfiler.Native/clr_helpers.cpp

```diff
@@ -311,6 +311,19 @@ mdAssemblyRef FindAssemblyRef(const ComPtr<IMetaDataAssemblyImport>& assembly_im
     return mdAssemblyRefNil;
 }
 
+mdAssemblyRef FindAssemblyRef(const ComPtr<IMetaDataAssemblyImport>& assembly_import, const shared::WSTRING& assembly_name, const Version& version)
+{
+    for (mdAssemblyRef assembly_ref : EnumAssemblyRefs(assembly_import))
+    {
+        auto assemblyMetadata = GetReferencedAssemblyMetadata(assembly_import, assembly_ref);
+        if (assemblyMetadata.name == assembly_name && assemblyMetadata.version == version)
```

Comment:
I was expecting the comparison to be `>=`

Reply:
I just validated this with a test and got the following error:
```
System.ArgumentException: 'The return type of the method: OnMethodBegin in type: Datadog.Trace.ClrProfiler.AutoInstrumentation.TraceAnnotations.TraceAnnotationsIntegration is not CallTargetState'
```

For an explanation of why this happens, for all of our CallTarget entrypoints we use the hard-coded version from the profiler. Say for instance that we have a .NET Framework application where the application references `2.6.0` but our automatic instrumentation is `2.5.0`, resulting in two `Datadog.Trace.dll` assemblies being loaded:
- The profiler creates TypeRefs for CallTarget types using the hard-coded version of `2.5.0`
- When creating a TypeRef for `TIntegration`, we must use the exact same assembly version of `2.5.0`. If we don't, the `CallTargetState`/`CallTargetReturn` types used inside the integration type will not match the types used in our IL Rewriting and there will be a casting error. For this reason, I must use the `==` comparison

I'll try to put some of these comments inside the code so the explanation is not lost

Reply:
Thank you for the clarification! Makes sense!.